src/engine/test/source/base/ipUtils_test.cpp

```diff
@@ -33,13 +33,94 @@ TEST(IPv4ToUInt, Invalid_range)
     EXPECT_THROW(utils::ip::IPv4ToUInt("1.256.1.1"), std::invalid_argument);
     EXPECT_THROW(utils::ip::IPv4ToUInt("1.1.256.1"), std::invalid_argument);
     EXPECT_THROW(utils::ip::IPv4ToUInt("1.1.1.256"), std::invalid_argument);
-
 }
 
 TEST(IPv4ToUInt, Valid_range)
 {
     EXPECT_EQ(utils::ip::IPv4ToUInt("0.0.0.0"), 0x0);
     EXPECT_EQ(utils::ip::IPv4ToUInt("127.0.0.1"), 0x7F'00'00'01);
-    EXPECT_EQ(utils::ip::IPv4ToUInt("192.168.0.1"), 0b11000000'10101000'00000000'00000001);
+    EXPECT_EQ(utils::ip::IPv4ToUInt("192.168.0.1"),
+              0b11000000'10101000'00000000'00000001);
     EXPECT_EQ(utils::ip::IPv4ToUInt("255.255.255.255"), 0xFFFFFFFF);
 }
+
+TEST(checkStrIsIPv4, Invalid_format)
+{
+    EXPECT_FALSE(utils::ip::checkStrIsIPv4(""));
+    EXPECT_FALSE(utils::ip::checkStrIsIPv4("1"));
+    EXPECT_FALSE(utils::ip::checkStrIsIPv4("1.2"));
+    EXPECT_FALSE(utils::ip::checkStrIsIPv4("1.2.3"));
+    EXPECT_FALSE(utils::ip::checkStrIsIPv4(" 1.2.3.4 "));
```

Comment:
shouldn't we discard empty spaces?

Reply:
No, it is possible that an IP can be used to concatenate with other values and you must necessarily have only the IP in the correct format (this is also an ECS requirement). In case you want to discard spaces you can use the trim function.